griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,302 @@
+            else:
+                return InfoArtifact(f"Google Doc '{file_path}' created with ID: {doc_id}")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"Error creating/saving Google Doc: {e}")
+
+    @activity(
+        config={
+            "description": "Can be used to load content from memory and save it to a new Google Doc "
+                           "in the specified folder.",
+            "schema": Schema(
+                {
+                    "memory_name": str,
+                    "artifact_namespace": str,
+                    "file_name": str,
+                    Optional(
+                        "folder_path",
+                        description="Path of the folder where the file should be saved.",
+                        default=DEFAULT_FOLDER_PATH,
+                    ): str,
+                }
+            ),
+        }
+    )
+    def save_memory_artifacts_to_google_docs(self, params: dict) -> ErrorArtifact | InfoArtifact:
+        values = params["values"]
+        memory = self.find_input_memory(values["memory_name"])
+    
+        if memory:
+            artifacts = memory.load_artifacts(values["artifact_namespace"])
+    
+            if artifacts:
+                try:
+                    file_path = values["file_name"]
+                    content = "\n".join([a.value for a in artifacts])
+
+                    save_params = {
+                        "file_path": file_path,
+                        "content": content,
+                        "folder_path": values.get("folder_path", self.DEFAULT_FOLDER_PATH),
+                    }
+
+                    return self.save_content_to_google_doc(save_params)
+    
+                except Exception as e:
+                    return ErrorArtifact(f"Error: {e}")
+    
+            else:
+                return ErrorArtifact("No artifacts found.")
+        else:
+            return ErrorArtifact("Memory not found.")
+
+    @activity(
+        config={
+            "description": "Can be used to download multiple Google Docs based on their paths.",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_paths",
+                        description="Destination file paths in the POSIX format. "
+                        "For example, 'foo/bar/baz.txt, foo/bar/baz2.txt'",
+                    ): [str]
+                }
+            ),
+        }
+    )
+    def download_google_docs(self, params: dict) -> ListArtifact | ErrorArtifact:
+        file_paths = params["file_paths"]
+        downloaded_files = []
+    
+        try:
+            service = self._build_client(self.DRIVE_SCOPES, "drive", "v3")
+    
+            for file_path in file_paths:
+                file_id = self._path_to_file_id(service, file_path)
+                if file_id:
+
+                    request = service.files().export_media(fileId=file_id, mimeType="text/plain")
```

Comment:
Where does this download the files to? Is there a way to specify an output directory?

Reply:
Nice catch! This should probably behave like `FileManager` with a custom `workdir`.

Reply:
So, we would essentially use the FileManger tool to save it to disk

Reply:
Actually, now that I thought about it, it's probably fine. Docs are downloaded into memory. If there is a `FileManager` available, the LLM can store those files to disk if necessary.